int64 position
int64 countPerSecond
float64 current
float64 h
float64 hdot